### DISCUSSION TITLE
How do you get active short term forecast text products?

### DISCUSSION THREAD

**Discussion by mbg206** [weather-gov/api#693] [Q&A]
I am looking to use the short term forecast text product in my project, but fetching them in the API (`https://api.weather.gov/products/types/NOW/locations/${wfo}`) returns a list of products where some are days old, and they don't have an expiration time. Is there a way to only get active products, or is there something that I am misunderstanding about how NWS text products work?

**Comment by JJKraw** [comment]
Is there a reason why just using the first (most recently issued) instance wouldn't work?  That's what I do.

**Reply by mbg206** [reply]
From my understanding, the short term forecasts are issued with the intention of only being valid [for the next few hours](https://w1.weather.gov/glossary/index.php?word=short+term+forecast), though when fetching products from some sites (like LWX), the most recent product is regarding morning fog on December 28, and I don't want to use days old short term forecasts in my project.

**Comment by dwhitemv25** [comment]
Some text products have an expiration, its the last token of the UGC string, if present. Take the first part of this non-precipitation warning message:

```
810
WWUS76 KPQR 050838
NPWPQR

URGENT - WEATHER MESSAGE
National Weather Service Portland OR
1238 AM PST Fri Jan 5 2024

ORZ005-006-WAZ022-039-051800-
/O.NEW.KPQR.FG.Y.0004.240105T0838Z-240105T1800Z/
Lower Columbia-Greater Portland Metro Area-I-
5 Corridor in Cowlitz County-Greater Vancouver Area-
Including the cities of St. Helens, Clatskanie, Hillsboro, 
Portland, Wilsonville, Oregon City, Gresham, Troutdale, Longview,
Kelso, Castle Rock, Vancouver, Battle Ground, Ridgefield, 
Washougal, Yacolt, and Amboy
1238 AM PST Fri Jan 5 2024

...DENSE FOG ADVISORY IN EFFECT UNTIL 10 AM PST THIS MORNING...
```

This line is the UGC. It specifies the zones covered by the message, and the message's expiration:
```
ORZ005-006-WAZ022-039-051800-
```

The `051800` is the expiration of the message (day-hour-minute UTC). Note that message expiration is different from the ending or expiration time of the warning itself (which is part of the VTEC string, the one that starts with `/O.`).

More information about UGC can be found in [NWS Instruction 10-1702](https://www.nws.noaa.gov/directives/sym/pd01017002curr.pdf).

**Reply by akrherz** [reply]
> Text products DO have an expiration,

Not all text products have UGC / expiration time :)  ie AFD

**Reply by dwhitemv25** [reply]
Updated comment, Mr. Pedantic. :)
